pyramda/iterable/flatten.py

```diff
@@ -0,0 +1,26 @@
+from collections import Iterable
+from past.builtins import basestring
+
+from pyramda.function.curry import curry
+
+
+@curry
+def flatten(xs, depth):
```

Comment:
I think your suggestion is cleaner . It gives the user the flexibility of specifying a leaf condition as opposed to my earlier implementation  where the basestring leaf condition is essentially hardcoded in the function implementation.

As far as depth is concerned,  are you suggesting that we don't support  the notion of recursion depth? So basically we keep recursing until there is nothing left to flatten?